Source/WebCore/platform/cocoa/ParentalControlsURLFilter.h

```diff
@@ -27,24 +27,27 @@
 
 #if HAVE(WEBCONTENTRESTRICTIONS)
 
+
 OBJC_CLASS WCRBrowserEngineClient;
 
 namespace WebCore {
 
 class ParentalControlsURLFilter {
     WTF_MAKE_FAST_ALLOCATED;
-    friend UniqueRef<ParentalControlsURLFilter> WTF::makeUniqueRefWithoutFastMallocCheck<ParentalControlsURLFilter>();
+    friend UniqueRef<ParentalControlsURLFilter> WTF::makeUniqueRefWithoutFastMallocCheck<ParentalControlsURLFilter>(const String&);
+
 public:
-    static ParentalControlsURLFilter& singleton();
+    static ParentalControlsURLFilter& filterWithConfigurationPath(const String&);
 
     bool isEnabled() const;
     void isURLAllowed(const URL&, CompletionHandler<void(bool, NSData *)>&&);
     void allowURL(const URL&, CompletionHandler<void(bool)>&&);
 
+    ParentalControlsURLFilter(const String& configurationPath);
```

Comment:
I know I shouldn't _need_ to make this initializer public, but calling `makeUniqueRef<ParentalControlsURLFilter>(key)` still complained about the use of a private constructor. I'm sure there's some combination of #includes I'm missing or some change I need to make to the `friend` declaration to make that work, but I couldn't figure it out.

Reply:
I think akin to this should work?

```
friend UniqueRef<NetworkExtensionContentFilter> WTF::makeUniqueRefWithoutFastMallocCheck<NetworkExtensionContentFilter>();
```

...that said, making it public is a viable option too.

Reply:
Yeah, I was trying to look at the NetworkExtensionContentFilter as a reference, and I think this looks the same, with the exception of the argument added to the constructor:
```
friend UniqueRef<ParentalControlsURLFilter> WTF::makeUniqueRefWithoutFastMallocCheck<ParentalControlsURLFilter>(const String&);
```

I also tried slapping some `WTF::`s on all the types, but that didn't seem to help. (I also tried some other things to make this class look more like NetworkExtensionContentFilter -- marking it `final`, making the class TZone  allocated instead of fast allocated, adding all the WTF imports that NetworkExtensionContentFilter.h has, but none of those were the missing ingredient.)